test/controller.test.ts

```diff
@@ -530,6 +501,367 @@ contract('Controller', ([owner, accountOwner1, accountOperator1, random]) => {
+          'Margin pool balance long otoken balance mismatch',
+        )
+        assert.equal(
+          receiverBalanceAfter.minus(receiverBalanceBefore).toString(),
+          collateralToWithdraw.multipliedBy(2).toString(),
+          'Receiver long otoken balance mismatch',
+        )
+        assert.equal(vaultAfter.longOtokens.length, 1, 'Vault long otoken array length mismatch')
+        assert.equal(
+          new BigNumber(vaultBefore.longAmounts[0]).minus(new BigNumber(vaultAfter.longAmounts[0])).toString(),
+          collateralToWithdraw.multipliedBy(2).toString(),
+          'Long otoken amount in vault after withdraw mismatch',
+        )
+      })
+
+      it('should remove otoken address from otoken array if amount is equal to zero after withdrawing', async () => {
+        const vaultCounter = new BigNumber(await controller.getAccountVaultCounter(accountOwner1))
+        assert.isAbove(vaultCounter.toNumber(), 0, 'Account owner have no vault')
+
+        const vaultBefore = await controller.getVault(accountOwner1, vaultCounter)
+
+        const collateralToWithdraw = new BigNumber(vaultBefore.longAmounts[0])
+        const actionArgs = [
+          {
+            actionType: ActionType.WithdrawLongOption,
+            owner: accountOwner1,
+            sender: accountOwner1,
+            asset: longOtoken.address,
+            vaultId: vaultCounter.toNumber(),
+            amount: collateralToWithdraw.toNumber(),
+            index: '0',
+            data: ZERO_ADDR,
+          },
+        ]
+        const marginPoolBalanceBefore = new BigNumber(await longOtoken.balanceOf(marginPool.address))
+        const receiverBalanceBefore = new BigNumber(await longOtoken.balanceOf(accountOwner1))
+
+        await controller.operate(actionArgs, {from: accountOwner1})
+
+        const marginPoolBalanceAfter = new BigNumber(await longOtoken.balanceOf(marginPool.address))
+        const receiverBalanceAfter = new BigNumber(await longOtoken.balanceOf(accountOwner1))
+        const vaultAfter = await controller.getVault(accountOwner1, vaultCounter)
+
+        assert.equal(
+          marginPoolBalanceBefore.minus(marginPoolBalanceAfter).toString(),
+          collateralToWithdraw.toString(),
+          'Margin pool balance long otoken balance mismatch',
+        )
+        assert.equal(
+          receiverBalanceAfter.minus(receiverBalanceBefore).toString(),
+          collateralToWithdraw.toString(),
+          'Receiver long otoken balance mismatch',
+        )
+        assert.equal(vaultAfter.longOtokens.length, 1, 'Vault long otoken array length mismatch')
+        assert.equal(vaultAfter.longOtokens[0], ZERO_ADDR, 'Vault long otoken address after clearing mismatch')
+        assert.equal(
+          new BigNumber(vaultBefore.longAmounts[0]).minus(new BigNumber(vaultAfter.longAmounts[0])).toString(),
+          collateralToWithdraw.toString(),
+          'Long otoken amount in vault after withdraw mismatch',
+        )
```

Comment:
add test to prevent withdrawing 0 options

Reply:
added.